packages/vc-verification/src/resolver/Ipfs-credential-resolver.ts

```diff
@@ -81,17 +86,22 @@ export class IpfsCredentialResolver implements CredentialResolver {
    *  provider,
    *  registrySettings,
    *  didStore );
-   * const credential = credentialResolver.getVerifiableCredential('did:ethr:1234', 'sampleRole');
+   * const credential = credentialResolver.getVerifiableCredential('did:ethr:1234', 'sampleRole', roleCredentialCache);
    * ```
    *
    * @param did subject DID for which the credential needs to be fetched
    * @param namespace role for which the credential needs to be fetched
+   * @param roleCredentialCache Cache to store role credentials
    * @returns
    */
-  async getVerifiableCredential(did: string, namespace: string) {
+  async getVerifiableCredential(
+    did: string,
+    namespace: string,
+    roleCredentialCache: IRoleCredentialCache
+  ) {
     const credentials = await this.credentialsOf(did);
     credentials.forEach((credential) =>
-      this._roleCredentialCache?.setRoleCredential(
+      roleCredentialCache.setRoleCredential(
```

Comment:
One thing to may consider here is is whether or not the addition of the `credentials` to the cache should not done in this function and instead was left to be done after this function returns.
This is because ideally, one should be able to trust that read function (which is implied by `get`) has no side-effects.
I think the side-effect to be avoided in this case is "[modifying a mutable argument passed by reference](https://en.wikipedia.org/wiki/Side_effect_(computer_science))"
What I'm thinking is that both the one credential that is requested plus the whole set of resolved credentials can be returned as a tuple. Then, the caller can add the set of credentials to the cache if they would like.

Reply:
So what you are referring is that we should get all credentials in the `getCredential` and then cache it? @jrhender

Reply:
@nichonien yes, what I'm envisioning is that `getVerifiableCredential` would be called like this
```
const cache = new CredentialCache();
const { namespaceCredential, allRetrievedCredentials } = getVerifiableCredential(did, namespace, cache);
cache.add(allRetrievedCredentials);
```
In this way, we remove the side-effect of cache modification from `getVerifiableCredential` and control/modification of the cache is fully in control of the caller. IMO, this approach is easier to reason about because there is no state change hidden in a method.

Reply:
Working at the code more, I think it is best to cache in `getVerifiableCredential` and `getEIP191JWT`. This is because the `getCredential` method is used only twice, while the specific methods are used for entire issuer's chain interaction. So `ClaimIssuerVerification` relies on `getEIP191JWT` for all credential resolution. What do you think @jrhender

Reply:
Ok, sounds good 👍

Reply:
Or maybe we do it in both the methods, so the `getCredential` relies on `credentialsOf` and `eip191JwtsOf` directly. So the credentials get cached within the main call method.

Reply:
You're suggestioning to add to the cache in `credentialsOf` and `eip191JwtsOf`?

Reply:
No, I was thinking to call `credentialsOf` and `eip191JwtsOf` from `getCredential` and then cache all the credentials. This way `getCredential`, `getVerifiableCredential` and `getEIP191JWT` would cache credentials whenever called, but this was increasing complexity.